### ISSUE TITLE
Regex searches

### ISSUE THREAD

**Issue by timholy** [JuliaComputing/JuliaHub-Feedback#133]
The "code" search page lets you use a regex, which is awesome. However, it doesn't specify which flavor of regex it supports. I made a few guesses and haven't come up with anything so far. My specific task is motivated by a problem I'm assigning as homework to a class: find a package that has method docstrings but lacks a module docstring, and contribute a module docstring using the soon-to-be-registered [ModuleDocstring package](https://github.com/JuliaDocs/ModuleDocstrings.jl) (see https://github.com/JuliaRegistries/General/pull/46228). This is essentially a warmup exercise on open-source contribution.

In Julia, I know the following regex finds "module" not preceded by a quote followed by a newline: `r"(?<!(\"\n))^module"m`. I've attempted several regex "flavors" on JuliaHub but not found anything that works. 

(Please add "search" label to this issue.)

**Comment by pfitzseb** [comment]
We're using https://github.com/google/codesearch in the backend, IIRC. So the issue is that we don't expose any way to provide flags (`m` in this case).

**Comment by pfitzseb** [comment]
No, this is a fair request, and we can definitely add a little bit of UI for regex flags (which I think are supported). A Julia based backend would also be cool, of course, but I don't think anyone on our end really has time for that at the moment.

**Comment by fonsp** [comment]
(That's a really cool homework exercise!)

**Comment by timholy** [comment]
Get 'em while they're fresh: https://github.com/timholy/AdvancedScientificComputing/blob/main/homeworks/pkgs_git_github.md
